modules/account-lib/src/coin/algo/utils.ts

```diff
@@ -89,48 +123,82 @@ export class Utils implements BaseUtils {
+    let decoded;
     try {
       decoded = this.decodeSeed(seed);
     } catch (e) {
       return false;
     }
 
-    let checksum = crypto.createHash('SHA512-256').update(decoded.seed).digest().slice(ALGORAND_NACL_SEED_BYTES_LENGTH - ALGORAND_CHECKSUM_BYTE_LENGTH, ALGORAND_NACL_SEED_BYTES_LENGTH);
-    return _.isEqual(decoded.checksum, checksum);
+    // Compute checksum
+    const checksum = new Uint8Array(
+      sha512.sha512_256.array(decoded.seed).slice(SEED_BYTES_LENGTH - ALGORAND_CHECKSUM_BYTE_LENGTH, SEED_BYTES_LENGTH),
+    );
+
+    // Check if the checksum and the seed are equal
+    return _.isEqual(checksum, decoded.checksum);
   }
 
   /**
-   * Decodes a seed
+   * encodeSeed decodes an algo seed
+   *
+   * Decoding algo seed is sane as decoding address.
+   * Latest version of algo sdk (1.9, at this writing) does not expose explicit method for decoding seed
+   * hence, this routine uses decodeAddress and changes the return structure
    *
-   * @param {string} seed - the seed to decode
-   * @returns {object} - a decoded seed that contains the seed and a checksum
+   * @param  {Buffer} secretKey - the secretKey valid.
+   * @returns {string} seed - the seed to be validated.
    */
-  protected decodeSeed(seed: string) {
-    if (typeof seed !== "string") throw ALGORAND_MALFORMED_SEED_ERROR; // try to decode
+  encodeSeed(secretKey: Buffer): string {
+    // return algosdk.encodeAddress(seed);
+    // get seed
+    const seed = secretKey.slice(0, SEED_BYTES_LENGTH);
+    // compute checksum
+    const checksum = sha512.sha512_256
+      .array(seed)
+      .slice(SEED_BYTES_LENGTH - ALGORAND_CHECKSUM_BYTE_LENGTH, SEED_BYTES_LENGTH);
+    const encodedSeed = base32.encode(concatArrays(seed, checksum));
 
-    const decoded = base32.decode.asBytes(seed); // Sanity check
+    return encodedSeed.toString().slice(0, ALGORAND_SEED_LENGTH); // removing the extra '===='
+  }
 
-    if (decoded.length !== ALGORAND_SEED_BYTE_LENGTH) throw ALGORAND_MALFORMED_SEED_ERROR;
+  /**
+   * decodeSeed decodes an algo seed
+   *
+   * Decoding algo seed is sane as decoding address.
+   * Latest version of algo sdk (1.9, at this writing) does not expose explicit method for decoding seed
+   * hence, this routine uses decodeAddress and changes the return structure
+   *
+   * @param {string} seed - the seed to be validated.
+   * @returns {Seed} - the object Seed
+   */
+  decodeSeed(seed: string): Seed {
+    // try to decode
+    const decoded = base32.decode.asBytes(seed);
+
+    // Sanity check
+    if (decoded.length !== ALGORAND_SEED_BYTE_LENGTH) throw new Error('seed seems to be malformed');
     return {
       seed: new Uint8Array(decoded.slice(0, ALGORAND_SEED_BYTE_LENGTH - ALGORAND_CHECKSUM_BYTE_LENGTH)),
-      checksum: new Uint8Array(decoded.slice(ALGORAND_NACL_SEED_BYTES_LENGTH, ALGORAND_SEED_BYTE_LENGTH)),
-    }
+      checksum: new Uint8Array(decoded.slice(SEED_BYTES_LENGTH, ALGORAND_SEED_BYTE_LENGTH)),
+    };
```

Comment:
Missing all these methods: https://github.com/BitGo/BitGoJS/compare/BG-34606-algo-platform-accountlib-migration#diff-1362121543d1f6184f7fa744d680bc2764d55791c8a7aa7b3eb3143395ebd333R459-R552

do we need them?

Reply:
Many of the methods that figured there were already implemented. In this PR were added just  isSeedValid, encodeSeed and decodeSeed.
https://github.com/BitGo/BitGoJS/commit/201085b74338fb3114752e41b3e4792668c39014#diff-1362121543d1f6184f7fa744d680bc2764d55791c8a7aa7b3eb3143395ebd333R394-R458